packages/@uppy/aws-s3/types/index.d.ts

```diff
@@ -10,8 +9,20 @@ export interface AwsS3UploadParameters {
     headers?: { [type: string]: string }
 }
 
-export interface AwsS3Options extends AwsS3MultipartOptions {
-    /** @deprecated future versions of this plugin will use the Expires value from the backend */
+export interface AwsS3Options {
+    id?: string
+    companionUrl?: string | null
+    companionHeaders?: Record<string, string>
+    allowedMetaFields?: Array<string> | null
+    getUploadParameters?: (file: UppyFile) => MaybePromise<{
+      method: "PUT" | "POST"
+      url: string
+      fields: Record<string, string> | null
```

Comment:
`fields` was a difficult one. On the one hand the docs say that the promise should ` fulfills with an object, with keys { method, url, fields, headers }.`  To `have the key` to me means it should not be allowed to be `undefined`.

However it also says that ` For presigned PUT uploads, this should be left empty.` 

I decided that `empty` but `defined` meant `null` was an option.

The docs say the other option was to return `an object with form fields`. `Record<string, string>` is almost certainly too strict; I would expect it to actually support something like `Record<string, string | Array<string> | number | Array<number>>` or even more. But I wanted to default into something as strict as possible over just using something like `Object`;

Reply:
I think a better type would be

```typescript
{
 method: 'POST'
 url: string
 fields: Record<string,string>
} | {
  method: 'PUT'
  url: string
}
```